estate/models/estate_property.py

```diff
@@ -48,3 +51,64 @@ def _default_date(self):
         'estate.property.tag',  # The related model (assuming this is your tag model)
         string='Tags'
     )
+    total_area = fields.Integer(compute = "_compute_total_area")
+    best_price = fields.Float(compute = "_compute_best_price")
+
+    # SQL Constraints
+    _sql_constraints = [
+        ("check_excepted_price", "CHECK(excepted_price > 0)", "A property expected price must be strictly positive."),
+        ("check_selling_price", "CHECK(excepted_price >= 0)", "A property selling price must be positive."),
+    ]
+
+    # Private methods
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for property in self:
+            property.best_price = max(property.offer_ids.mapped("price")) if property.offer_ids else 0
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        for property in self:
+            if property.garden :
+                property.garden_area = property.garden * 2
+                property.garden_orientation = "north"
+            else :
+                property.garden_area = 0
+                property.garden_orientation = False
+
+    @api.onchange("date_availability")
+    def _onchange_date_availability(self):  
+        for property in self:
+            if (property.date_availability - fields.Date.today()).days < 0 :
+                return {
+                    "warning": {
+                        "title": ("Warning"),
+                        "message": ("The availability date is set to a date prior to today.")
+                    }
+                }
+            
+    # Python constraints
+    @api.constrains("selling_price", "excepted_price") # this allow triggered the constraint every time selling price or the expected price is changed
+    def _check_constraints(self):
+        for property in self:
+            if property.selling_price < 0.9* property.excepted_price:
+                raise ValidationError(("The selling price cannot be lower than 90 percent of the expected_price"))
+                
+    # Public methods
+    # methods for sold and canceled property buttons
+    def action_sold(self):
+        for property in self:
```

Comment:
Pas de loop dans une action
```suggestion
        self.ensure_one()
```